lib/buffer.js

```diff
@@ -593,6 +593,15 @@ Buffer.concat = function concat(list, length) {
     validateOffset(length, 'length');
   }
 
+  // If there is only one FastBuffer, rotate without copying
+  if (
+    list.length === 1 &&
+    list[0] instanceof FastBuffer &&
+    list[0].length === length
+  ) {
+    return Buffer.from(list[0]);
```

Comment:
If `list[0]` is a TypedArray (like another `Buffer`) then this still copies.

Reply:
I’m wondering if it makes sense to return list[0] directly. I understand this avoids copying, so I guess I should update the test accordingly?
https://github.com/nodejs/node/blob/main/test/parallel/test-buffer-concat.js#L43

Reply:
`Buffer.concat` should always return a new buffer object: https://github.com/nodejs/node/blob/91d2400c74a5d5e9b57bd6ca861fe56ff9b7f872/doc/api/buffer.md?plain=1#L1033-L1034

Reply:
As I understand it, at the end of the day we should always create a buffer.